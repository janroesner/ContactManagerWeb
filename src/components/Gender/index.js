import React, { PropTypes } from 'react'

const Gender = ({gender}) => {
  return(
    <span>
      { gender === 'm' ? 'ğŸ‘¨' : 'ğŸ‘©' }
    </span>
  )
}

// Note: here we should check the prop for being 'm' or 'f' in addition
Gender.propTypes = {
  gender: PropTypes.string.isRequired
}

export default Gender
