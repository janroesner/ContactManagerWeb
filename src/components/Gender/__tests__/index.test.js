import React from 'react';
import Gender from '../';
import renderer from 'react-test-renderer';

test('renders male emoji for males', () => {

  const component = renderer.create(
    <Gender gender='m' />
  )
  let tree = component.toJSON()

  expect(tree).toMatchSnapshot()
})

test('renders female emoji for females', () => {

  const component = renderer.create(
    <Gender gender='f' />
  )
  let tree = component.toJSON()

  expect(tree).toMatchSnapshot()
})

test('render female emoji as default, when invalid gender is given', () => {

  const component = renderer.create(
    <Gender gender='a' />
  )
  let tree = component.toJSON()

  expect(tree).toMatchSnapshot()
})
