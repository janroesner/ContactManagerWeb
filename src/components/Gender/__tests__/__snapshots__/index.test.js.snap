// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render female emoji as default, when invalid gender is given 1`] = `
<span>
  ðŸ‘©
</span>
`;

exports[`renders female emoji for females 1`] = `
<span>
  ðŸ‘©
</span>
`;

exports[`renders male emoji for males 1`] = `
<span>
  ðŸ‘¨
</span>
`;
